src/interfaces/file/input.vue

```diff
@@ -109,40 +93,50 @@ export default {
     };
   },
   computed: {
+    file() {
+      if (this.value !== null) return this.value;
+      if (this.values !== null && this.values.filesize && this.values.filename) return this.values;
+      return void 0;
+    },
+    filePreviewOptions() {
+      if (!this.options || !this.options.edit) return {};
+      return {
+        remove: {
+          text: this.$t("delete"),
+          icon: "delete"
+        }
+      };
+    },
     subtitle() {
-      if (!this.value) return "";
+      if (!this.file) return "";
 
       return (
-        this.value.filename.split(".").pop() +
+        this.file.filename.split(".").pop() +
         " • " +
-        this.$d(new Date(this.value.uploaded_on), "short")
+        this.$d(new Date(this.file.uploaded_on), "short")
       );
     },
     subtitleExtra() {
       // Image ? -> display dimensions and formatted filesize
-      return this.value.type && this.value.type.startsWith("image")
+      return this.file.type && this.file.type.startsWith("image")
         ? " • " +
-            this.value.width +
+            this.file.width +
             " x " +
-            this.value.height +
+            this.file.height +
             " (" +
-            formatSize(this.value.filesize) +
+            formatSize(this.file.filesize) +
             ")"
         : null;
     },
     src() {
-      return this.value.type && this.value.type.startsWith("image")
-        ? this.value.data.full_url
-        : null;
+      return this.file.type && this.file.type.startsWith("image") ? this.file.data.full_url : null;
```

Comment:
Lets refactor this into something more verbose and legible, for example:

```suggestion
      const fileType = this.file.type;
      
      if (!fileType) return null;
      
      const isImage = fileType.startsWith("image");
      
      if (isImage) return this.file.data.full_url;
      
      return null;
```